app/controllers/openid_connect/logout_controller.rb

```diff
@@ -7,16 +7,22 @@ class LogoutController < ApplicationController
     include OpenidConnectRedirectConcern
 
     before_action :set_devise_failure_redirect_for_concurrent_session_logout,
-                  only: [:logout]
+                  only: [:show, :create]
     before_action :confirm_two_factor_authenticated, only: [:delete]
+    skip_before_action :verify_authenticity_token, only: [:create]
 
-    # Handle logout (with confirmation if initiated by relying partner)
-    def logout
+    # +GET+ Handle logout (with confirmation if initiated by relying partner)
+    # @see {OpenID Connect RP-Initiated Logout 1.0 Specification}[https://openid.net/specs/openid-connect-rpinitiated-1_0.html#RPLogout]  # rubocop:disable Layout/LineLength
+    def show
       @logout_form = build_logout_form
       result = @logout_form.submit
       redirect_uri = result.extra[:redirect_uri]
-
-      analytics.oidc_logout_requested(**to_event(result))
+      extra_event_props = session[:event] || {}
```

Comment:
if there's only one key, does it need to be a hash?

Reply:
I suppose not. I just didn't want to have collisions with other keys in the session getting populated by callbacks and concerns.

Reply:
I actually thought making the field `referrer_method`, but figured that might be confusing as to how that information was sourced.

Reply:
Changed to set directly on `session`. 🤞🏽 this doesn't overwrite anything!